src/ecore.js

```diff
@@ -861,6 +868,9 @@ EClass.values = {
 
         return _.isArray(subTypes) ? subTypes : [];
     },
+    hasSubTypes: function () {
```

Comment:
Maybe what you are looking for is to implement an equivalent to what is done in the Java version, see https://github.com/eclipse/emf/blob/db2f900cb6ce9f1e184d60da11d92f650f2840be/plugins/org.eclipse.emf.ecore.xmi/src/org/eclipse/emf/ecore/xmi/impl/XMLSaveImpl.java#L3677